src/Avalonia.Base/Media/Fonts/Tables/Name/NameRecord.cs

```diff
@@ -2,44 +2,58 @@
 // Licensed under the Apache License, Version 2.0.
 // Ported from: https://github.com/SixLabors/Fonts/blob/034a440aece357341fcc6b02db58ffbe153e54ef/src/SixLabors.Fonts
 
+using System;
+
 namespace Avalonia.Media.Fonts.Tables.Name
 {
-    internal class NameRecord
+    internal readonly struct NameRecord
     {
-        private readonly string value;
-
-        public NameRecord(PlatformIDs platform, ushort languageId, KnownNameIds nameId, string value)
+        private readonly ReadOnlyMemory<byte> _stringStorage;
+
+        public NameRecord(
+            ReadOnlyMemory<byte> stringStorage,
+            PlatformID platform,
+            ushort languageId,
+            KnownNameIds nameId,
+            ushort offset,
+            ushort length,
+            System.Text.Encoding encoding)
         {
+            _stringStorage = stringStorage;
+
             Platform = platform;
             LanguageID = languageId;
             NameID = nameId;
-            this.value = value;
+            Offset = offset;
+            Length = length;
+            Encoding = encoding;
         }
 
-        public PlatformIDs Platform { get; }
+        public PlatformID Platform { get; }
 
         public ushort LanguageID { get; }
 
         public KnownNameIds NameID { get; }
 
-        internal StringLoader? StringReader { get; private set; }
+        public ushort Offset { get; }
 
-        public string Value => StringReader?.Value ?? value;
+        public ushort Length { get; }
 
-        public static NameRecord Read(BigEndianBinaryReader reader)
+        public System.Text.Encoding Encoding { get; }
+
+        public string Value
```

Comment:
Should probably be a method, as a property it will be easy to use several times in a row without realizing that it allocates every time.